Create panneau_electoral.yml

pdfext opened the pull request:
Tentative de création d'un profil 'paneau_electoral' pour permettre la contribution sur les panneaux éléctoraux dans OSM à travers 'OSM contributor'

pdfext committed:
Create panneau_electoral.yml

Tentative de création d'un profil 'paneau_electoral' pour permettre la contribution sur les panneaux éléctoraux dans OSM à travers 'OSM contributor'

loicortola commented:
Bonjour,

merci beaucoup pour ce super preset.
Les conventions étant de nommer les fichiers en anglais, merci de renommer ce fichier en political_board.yml

D'autre part, il y a quelques erreurs de syntaxe sur le fichier qui m'empêchent de merger en l'état:
le premier champ s'appelle "ame" et non "name". Merci de corriger la typo.

Le YAML est un langage qui tient compte de l'importance de l'indentation.
Certains éléments ont une indentation mal réglée (quel est votre éditeur de texte?), et je vous conseille de valider sa structure avec un validateur de syntaxe yaml (http://www.yamllint.com/)[http://www.yamllint.com/]

Voici la trace d'erreur générée par notre parser de preset:
{ YAMLException: bad indentation of a mapping entry at line 18, column 6:
         icon: 'https://commons.wikimedia ...
         ^
    at generateError (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:162:10)
    at throwError (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:168:9)
    at readBlockMapping (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:1074:7)
    at composeNode (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:1326:12)
    at readBlockSequence (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:924:5)
    at composeNode (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:1325:12)
    at readBlockMapping (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:1056:11)
    at composeNode (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:1326:12)
    at readDocument (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:1488:3)
    at loadDocuments (/Users/loicortola/workspace/h2geo-presets/node_modules/js-yaml/lib/js-yaml/loader.js:1544:5)
  name: 'YAMLException',
  reason: 'bad indentation of a mapping entry',
  mark:
   Mark {
     name: null,
     buffer: 'name:\n  default: Preset Panneau Electoral\n  en: Political Board Preset\n  fr: Preset Panneau Electoral\nversion: \'0.1\'\nauthor: PaDe\nlink: https://github.com/jawg/h2geo-presets\ndescription:\n  default: Pour contribuer au sujet des Panneaux Electoraux\n  en: To contribute on Political Boards\n  fr: Pour contribuer au sujet des Panneaux Electoraux\nimage: \'https://commons.wikimedia.org/wiki/File:PanneauElectoral_20170423.jpg\'\ngroups:\n  - name:\n      default: Advertising\n      en: Advertising\n      fr: Publicité\n     icon: \'https://commons.wikimedia.org/wiki/File:PanneauElectoral_20170423.jpg\'\n     url: \'https://wiki.openstreetmap.org/wiki/Category:Advertising\'\n     items:\n      - name: advertising=board\n        url: \'https://wiki.openstreetmap.org/wiki/FR:Tag:advertising%3Dboard\'\n        label:\n          default: Advertising board\n          en: Advertising board\n          fr: Panneau électoral\n        description:\n          default: Advertising board for political advertising during electoral campaign\n          en: Advertising board for political advertising during electoral campaign\n          fr: Panneau électoral, pour les candidats aux élections\n        keywords:\n          default:\n            - Advertising\n            - Political board\n          en:\n            - Advertising\n            - Political board\n          fr:\n            - Publicité\n            - Panneau electoral\n        tags:\n          - key: advertising\n            value: board\n            type: CONSTANT\n          - key: permanent\n            value : no\n            type: CONSTANT\n          - key: message\n            value : political\n            type: CONSTANT\n\u0000',
     position: 513,
     line: 17,
     column: 5 },
  message: 'bad indentation of a mapping entry at line 18, column 6:\n         icon: \'https://commons.wikimedia ... \n         ^' }

Je suis à votre écoute si besoin d'un coup de main, mais les changements à faire sont assez simples.

pdfext committed:
Update panneau_electoral.yml

1) Fixed typo
2) Validated the syntax using http://www.yamllint.com/, and used the output of yamllint  as it is a "nice clean UTF-8 version of it"
3) Put É and é where they were wrongly E and e in french

pdfext committed:
Rename panneau_electoral.yml to political_board.yml

All is said in summary

loicortola commented:
Super, il reste juste une valeur à changer mais je m'en occupe!
